metricbeat/helper/metricset.go

```diff
@@ -52,12 +52,19 @@ func (m *MetricSet) Fetch() error {
 		go func(h string) {
 			defer m.Module.wg.Done()
 
+			starttime := time.Now()
 			event, err := m.MetricSeter.Fetch(m, h)
+			endtime := time.Now()
```

Comment:
`elapsed := time.Since(starttime)` is the shorthand version of what you are doing with `time.Sub()`.

I wish Go had a precision clock for timing purposes (like Java's `System.nanoTime()`). By using wall-clock time your measurements are affected by clock changes (i.e. if NTP adjusts the wall-clock then your measurement is affected).